google/cloud/bigtable/data/execute_query/_parameters_formatting.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, Any, Optional
+import datetime
+from google.api_core.datetime_helpers import DatetimeWithNanoseconds
+from google.cloud.bigtable.data.exceptions import ParameterTypeInferenceFailed
+from google.cloud.bigtable.data.execute_query.values import ExecuteQueryValueType
+from google.cloud.bigtable.data.execute_query.metadata import SqlType
+
+
+def _format_execute_query_params(
+    params: Optional[Dict[str, ExecuteQueryValueType]],
+    parameter_types: Optional[Dict[str, SqlType.Type]],
+):
+    if not params:
```

Comment:
docstrings on these would also be helpful for future maintenance

It doesn't need to be too complex for the internal functions, but some context would be useful

Reply:
done